Carta.py

```diff
@@ -343,67 +357,153 @@ def opponentRespondsInGrabPhase(self):
         # if the time pass start time is 18 seconds and it checks that there is no grabbing card match the last word of the currentReadingcard,
-        # and opponent didn't grabbed any grabbing card, but you garbbed a card, then opponent wins
+        # and opponent didn't grabbed any grabbing card, but you grabbed a card, then opponent wins
         elif ((self.GPinfo.timesUp is True) and \
-              (self.checkGrabbingAvailable() is False) and \
-              (self.GPinfo.yourGrabCardLastWord != "") and \
-              (self.GPinfo.oppoGrabCardLastWord == "")):
+              (self.GPinfo.correctGrabCardStatus is GrabCardStatus.INVALID) and \
+              (self.GPinfo.yourStatus is not GrabPhaseStatus.NO_TOUCH) and \
+              (self.GPinfo.oppoStatus is GrabPhaseStatus.NO_TOUCH)):
             self.GPinfo.youWin = False
             self.GPinfo.opponentWin = True
             statement = "Carta.py: decideWinner: No available GrabCard, opponent win"
+            self.phase = Phase.OPPONENT_TRANSFER
+            self.numCardsToTransfer += 1
         # if the time pass start time is 18 seconds and it checks that there is no grabbing card match the last word of the currentReadingcard,
         # and opponentgrabbed a grabbing card, but you didn't grabbed any card, then you wins
         elif ((self.GPinfo.timesUp is True) and \
-              (self.checkGrabbingAvailable() is False) and \
-              (self.GPinfo.yourGrabCardLastWord == "") and \
-              (self.GPinfo.oppoGrabCardLastWord != "")):
+              (self.GPinfo.correctGrabCardStatus is GrabCardStatus.INVALID) and \
+              (self.GPinfo.yourStatus is GrabPhaseStatus.NO_TOUCH) and \
+              (self.GPinfo.oppoStatus is not GrabPhaseStatus.NO_TOUCH)):
             self.GPinfo.youWin = True
             self.GPinfo.opponentWin = False
             statement = "Carta.py: decideWinner: No available GrabCard, you win"
+            self.phase = Phase.YOUR_TRANSFER
+            self.numCardsToTransfer += 1
         else:
             self.GPinfo.youWin = False
             self.GPinfo.opponentWin = False
             statement = "Carta.py: decideWinner: Both players grab wrong card"
+
+        self.appendDialogsWhenDecidingWinner()
+        self.usedReadingCards.cards.append(self.curReadingCard)
+        self.removeCorrectGrabbingCard()
+
         if (self.debugMode):
             print(statement)
-        self.phase = Phase.OPPONENT_TRANSFER
+
+        if ((self.numYourGrabCardInPlay == 0) or (self.numOppoGrabCardInPlay == 0)):
+            self.phase = Phase.END
+        else:
+            self.GPinfo.reset()  # Cannot be a phase because it might overlap with Transfer phase
+        self.displayReadingCard = False
+
+    # Not validated
+    def opponentTransfers(self):
+        for numberTransferred in range(self.numCardsToTransfer):
+            random.seed(time.time())
+            x = random.randint(1, self.numOppoGrabCardInPlay)
```

Comment:
not assign to a random place, but put that card on the right of your field. 

At YOUR_SET_UP phase, move the card to your field.